editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -281,45 +281,49 @@ export const UiJsxCanvas = betterReactMemo(
 
     // TODO after merge requireFn can never be null
     if (requireFn != null) {
-      let resolvedFiles: string[] = []
+      let resolvedFiles: MapLike<Array<string>> = {} // Mapping from importOrigin to an array of toImport
       const customRequire = React.useCallback(
         (importOrigin: string, toImport: string) => {
-          const filePathResolveResult = resolve(importOrigin, toImport)
+          if (resolvedFiles[importOrigin] == null) {
+            resolvedFiles[importOrigin] = []
+          }
+          let resolvedFromThisOrigin = resolvedFiles[importOrigin]
+
+          const alreadyResolved = resolvedFromThisOrigin.includes(toImport) // We're inside a cyclic dependency, so trigger the below fallback
+          const filePathResolveResult = alreadyResolved
+            ? left<string, string>('Already resolved')
```

Comment:
so what I _still_ don't understand is why can't we just return the already resolved (and cached) result here?